### PR TITLE
Wi-Fi upmerge for 2.8.0

### PR DESCRIPTION
**Pull request by krish2718**
This cherry-picks all Wi-Fi fixes (Boards, drivers and hostap) for 2.8.0 release.

### CODE REVIEW

**Review by bama-nordic — APPROVED** [review]

**Review by srkanordic — APPROVED** [review]

**Review by kapbh — APPROVED** [review]

**Review by rado17 — APPROVED** [review]

**Review by sachinthegreen — APPROVED** [review]

File: include/zephyr/settings/settings.h [hunk 1]
```diff
@@ -36,7 +36,7 @@ extern "C" {
 
 #define SETTINGS_MAX_DIR_DEPTH	8	/* max depth of settings tree */
 #define SETTINGS_MAX_NAME_LEN	(8 * SETTINGS_MAX_DIR_DEPTH)
-#define SETTINGS_MAX_VAL_LEN	256
+#define SETTINGS_MAX_VAL_LEN	512
```
Comment (rlubos) [inline include/zephyr/settings/settings.h:39]: Why is this a noup? Shouldn't this be addressed upstream instead?

**Review by rlubos — COMMENTED** [review]
Please don't come so lightly to noup patches, we don't have automated tools to track/revert them so it's not so easy to reduce the technical debt later on.

File: include/zephyr/settings/settings.h (hunk 1 above)
Comment (krish2718) [reply include/zephyr/settings/settings.h:39]: @rado17  please raise a PR upstream

File: boards/nordic/nrf7002dk/nrf7002dk_nrf5340_cpuapp_nrf7001_ns.dts [hunk 2]
```diff
@@ -35,6 +35,5 @@
 		qspi-quad-mode;
 
 		#include "nrf70_common.dtsi"
-		#include "nrf70_common_coex.dtsi"
```
[the remark this answers is not in the document]
Comment (krish2718) [reply boards/nordic/nrf7002dk/nrf7002dk_nrf5340_cpuapp_nrf7001_ns.dts:38]: there isn't, NS boards are removed upstream, so, this has to be noup.

File: modules/hostap/CMakeLists.txt [hunk 3]
```diff
@@ -678,8 +678,4 @@ if(CONFIG_SAE_PWE_EARLY_EXIT)
 	"please use it carefully and do not use it production.")
 endif()
 
-zephyr_library_compile_definitions_ifdef(CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ALT
```
[the remark this answers is not in the document]
Comment (krish2718) [reply modules/hostap/CMakeLists.txt:518]: All these `*NCS*` configs are purely for NCS i.e, they use nRF security. I have a better alternative that I will work post-2.8.0 that removed most of these `noup` patches related to `hostap` config management.

File: samples/net/wifi/Kconfig.sysbuild [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config WIFI_NRF70
```
[the remark this answers is not in the document]
Comment (krish2718) [reply samples/net/wifi/Kconfig.sysbuild]: I will fix this.

File: samples/net/wifi/sample.yaml [hunk 5]
```diff
@@ -65,8 +65,8 @@ tests:
       - nucleo_h723zg
   sample.net.wifi.nrf7002eb:
     extra_args:
-      - NRF70_UTIL=y
       - SB_CONFIG_WIFI_NRF70=y
+      - CONFIG_NRF70_UTIL=y
```
[the remark this answers is not in the document]
Comment (krish2718) [reply samples/net/wifi/sample.yaml:69]: I will fix this.

File: include/zephyr/settings/settings.h (hunk 1 above)
Comment (krish2718) [reply include/zephyr/settings/settings.h:39]: I will drop this from this PR, we can raise a fix separately, don't want to hold this PR for this.

File: samples/net/wifi/Kconfig.sysbuild (hunk 4 above)
[the remark this answers is not in the document]
Comment (krish2718) [reply samples/net/wifi/Kconfig.sysbuild]: Actually, as it's noup it was causing conflicts, so, had to revert original fix and reapply as noup again :)

**Review by rlubos — APPROVED** [review]

**Comment by rlubos** [comment]
Commit tag check fails because we could not pull one of the dependencies causing conflict in the manifest file (https://github.com/nrfconnect/sdk-zephyr/commit/942a122fce5df5dbc7ccd5d449478704e2b013ef#diff-d7f4082837dc3cb270fcf1abd68188216fdadee294e01611b8ca59ddea7fdbb3R191) as it broke low-level tests.
I'm opting for CI exception for this particular case. CC @carlescufi

**Pull request merged**